core/span_enrichment_test.go

```
@@ -0,0 +1,117 @@
+	schemas.AttrBifrostRoutingEngineUsed: "framework ExtractedFields: framework/tracing/tracer.go",
+}
+
+// TestContextSpanAttributesEmit drives applyContextSpanAttributes with every
+// context key populated and asserts each dimension lands on the span under its
+// canonical SpanAttr with the exact value. A dead emit (a dim dropped from the
+// helper) fails as a missing attribute; a mis-wired read (wrong context key)
+// fails as a wrong/missing value, since every sentinel is distinct.
+func TestContextSpanAttributesEmit(t *testing.T) {
+	ctx := context.Background()
+	for _, d := range contextDimSources {
+		ctx = context.WithValue(ctx, d.ctxKey, d.value)
```

golangci-lint (2.13.2)

[error] nested context in loop

(fatcontext)